client/lib/signup/step-actions/index.js

```diff
@@ -606,9 +607,9 @@ async function addExternalManagedThemeToCart( state, dispatch, themeId, siteSlug
 export function addPlanToCart( callback, dependencies, stepProvidedItems, reduxStore ) {
 	// Note that we pull in emailItem to avoid race conditions from multiple step API functions
 	// trying to fetch and update the cart simultaneously, as both of those actions are asynchronous.
-	const { addOnItem, emailItem, siteSlug, plugin, billing_period: billingPeriod } = dependencies;
-	const { cartItem, lastKnownFlow } = stepProvidedItems;
-	if ( isEmpty( cartItem ) && isEmpty( emailItem ) ) {
+	const { emailItem, siteSlug, plugin, billing_period: billingPeriod } = dependencies;
+	const { cartItems, lastKnownFlow } = stepProvidedItems;
+	if ( cartItems.some( ( item ) => item === null ) && isEmpty( emailItem ) ) {
```

Comment:
**Note:**

I'm not using isEmpty because of broader efforts to deprecate `lodash` https://calypsop2.wordpress.com/2021/02/24/the-path-to-deprecating-lodash/